### DISCUSSION TITLE
Is there any way to receive the old sent message before subscriber init?

### DISCUSSION THREAD

**Discussion by RockyXRQ** [eclipse-ecal/ecal#1856] [Q&A] [answered]
I'm using callback mode to receive the message. For everytime my publisher sent a message before the subscriber init, my subscriber can't receive it anymore. Do we have any ways to achieve it?

**Comment by DownerCase** [comment, marked as answer]
As far I know - No. If the publisher only sends messages to the subscribers it is aware of. I'd recommend setting up any subscribers and publishers your program will need as early as possible.

You could consider a synchronization delay between initializing the publisher/subscribers and sending the data; [decreasing the registration refresh period](https://eclipse-ecal.github.io/ecal/configuration/options.html#cmdoption-arg-registration_refresh), or using the client-server service API (if you need reliable call and response).

**Reply by RockyXRQ** [reply]
Thank u so much for the reply! I just switched to client-server service API and it works perfect!

**Comment by rex-schilasky** [comment]
Thank you @DownerCase. That fits all.
